Sources/TuistGenerator/Utils/SettingsHelper.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+import XcodeProj
+
+final class SettingsHelper {
+    func extend(buildSettings: inout [String: Any], with other: [String: Any]) {
```

Comment:
Why are the settings `[String: Any]`? I assume making it `[String: String]` would cause a bug?

Reply:
`[String: String]` would make the comparison much easier but `Any` type comes from xcodeproj as it's used in the `BuildSettingsProvider`. https://github.com/tuist/xcodeproj/blob/8fb841d76bdcc47ce2936a85e7d3d31da14eafd0/Sources/xcodeproj/Utils/BuildSettingsProvider.swift#L236

It would be great to simplify that bit! I will check if xcodeproj relays on that data structure in any way (except `BuildSettingsProvider`). 

@pepibumur any thoughts?

Reply:
If I'm not wrong we can change the type to `[String: String]` in XcodeProj

Reply:
@pepibumur great news! Do you guys think I can do the `[String: Any]` -> `[String: String]` refactoring as a follow up (in a separated PR)? I'd like to avoid making that diff bigger.

Reply:
Sounds like a sensible idea!

Reply:
Agree